Adjust path to gallery to support android 10 below

jollyash opened the pull request:
Merges recent Android <= 10 devices. Been through and passed QA. 

There are probably some nicer ways / clear up in places - so could you give it a once over and comment anywhere you think needs looking at.

jollyash committed:
Adjust path to gallery to support android 10 below

phillip-hogan commented on src/android/SaveImageGallery.java:
```diff
@@ -167,39 +175,76 @@ private File savePhoto(Bitmap bmp, String prefix, String format, int quality) {
+              values.put(Images.Media.DATE_ADDED, System.currentTimeMillis() / 1000);
+              values.put(Images.Media.DATE_TAKEN, System.currentTimeMillis());
+
+              url = this.cordova.getActivity().getApplicationContext().getContentResolver().insert(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, values);
```
Could you possibly put the call to `this.cordova.getActivity().getApplicationContext().getContentResolver()` into a variable and then reuse it rather than needing to call it twice?

phillip-hogan commented on src/android/SaveImageGallery.java:
```diff
@@ -167,39 +175,76 @@ private File savePhoto(Bitmap bmp, String prefix, String format, int quality) {
+              OutputStream imageOut = this.cordova.getActivity().getApplicationContext().getContentResolver().openOutputStream(url);
+
+              try {
+                if (format.equalsIgnoreCase(JPG_FORMAT)) {
```
Do we need this if statement if the default is JPEG any way? Could it not just be:

`if(format.equalsIgnoreCase(PNG_FORMAT))
{...}
else { the JPEF default}`

jollyash replied:
I just copied this if statement from the original code. 

The first check is for JPG format. So perhaps it's meant to add the .jpg to the file name, but i'm not sure if compressFormat.JPG is a thing or if it just uses the compressFormat.JPEG in the case of a jpg.

I might be overthinking it.

phillip-hogan replied:
Ah ok, in that case just leave it as is.

phillip-hogan reviewed (approved):
Looks good @jollyash! 

Two very minor comments that I'll leave up to you if you want to address or not.

jollyash committed:
PR changes

- Place content resolver inside variable
- Slight change to file exstenion name

jollyash commented on src/android/SaveImageGallery.java:
```diff
@@ -167,6 +167,7 @@ private void saveBase64Image(JSONArray args, CallbackContext callbackContext) th
      */
     private File savePhoto(Bitmap bmp, String prefix, String format, int quality) {
         File retVal = null;
+        ContentResolver contextResolver = this.cordova.getActivity().getApplicationContext().getContentResolver();
```
@phillip-hogan Good shout. ContentResolver moved inside a variable
